test/test_testing.py

```diff
@@ -2748,6 +2755,27 @@ def test_add_network_all_args(self):
         self.assertEqual(interface.address, ipaddress.IPv4Address('10.0.0.10'))
         self.assertEqual(interface.subnet, ipaddress.IPv4Network('10.0.0.0/8'))
 
+    def test_add_network_specific_endpoint(self):
+        self.harness.add_network('10.0.0.1')
+        self.harness.add_network('10.0.2.1', endpoint='db')
+
+        binding = self.harness.model.get_binding('db')
+        self.assertEqual(binding.name, 'db')
+        network = binding.network
+        self.assertEqual(network.bind_address, ipaddress.IPv4Address('10.0.2.1'))
+
```

Comment:
The other thing that you could check in both this test and the next, is that the binding for 'foo' is still on the default value. To make it clear that the change here does impact one relation/endpoint but not all of them.

Reply:
Good call - done.